apps/monarch/monarch-frontend/src/app/SearchTherapists.tsx

```diff
@@ -65,46 +65,249 @@ const debouncedSearchTherapists = debouncePromise(
+    setFullName(therapist.fullName);
+    setAddress(therapist.address);
+    setEmail(therapist.email);
+    setWebsite(therapist.website);
+    setFamiliesHelped(0);
+  };
+
+  const handleClose = () => {
+    resetForm();
+    onEditClose();
+  };
+
+  const handleSave = () => {
+    if (isFormValid()) {
+      // TODO: save data in database
+      onEditClose();
+    }
+  };
+
+  return (
+    <>
+      <Button onClick={onEditOpen} colorScheme="teal" size="sm">
+        Edit
+      </Button>
+
+      <Modal isOpen={isEditOpen} onClose={onEditClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Edit Practitioner Info</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <VStack spacing={4}>
+              <FormControl isRequired>
+                <FormLabel>Phone</FormLabel>
+                <Input
+                  value={phone}
+                  onChange={(e) => setPhone(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Full Name</FormLabel>
+                <Input
+                  value={fullName}
+                  onChange={(e) => setFullName(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Address</FormLabel>
+                <Input
+                  value={address}
+                  onChange={(e) => setAddress(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Email</FormLabel>
+                <Input
+                  value={email}
+                  onChange={(e) => setEmail(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Date Joined (YYYY-MM-DD)</FormLabel>
+                <Input
+                  value={dateJoined}
+                  onChange={(e) => setDateJoined(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Families Helped</FormLabel>
+                <Input
+                  value={familiesHelped}
+                  onChange={(e) => {
+                    const value = e.target.value;
+                    if (value === '' || /^[0-9]+$/.test(value)) {
+                      setFamiliesHelped(value === '' ? 0 : Number(value));
+                    }
+                  }}
+                  onKeyDown={(e) => {
+                    if (
+                      e.key === 'Backspace' ||
+                      e.key === 'Delete' ||
+                      e.key === 'ArrowLeft' ||
+                      e.key === 'ArrowRight' ||
+                      /^[0-9]$/.test(e.key)
+                    ) {
+                      return;
+                    }
+                    e.preventDefault();
+                  }}
+                />
```

Comment:
This is great, but can we use [chakra's number input components](https://v2.chakra-ui.com/docs/components/number-input/usage) here instead to keep things simple? It should take care of only allowing numeric input, and we can get increment/decrement buttons easily as well